homeassistant/components/airvisual_pro/sensor.py

```diff
@@ -0,0 +1,149 @@
+        name="PM 2.5",
+        device_class=SensorDeviceClass.PM25,
+        native_unit_of_measurement=CONCENTRATION_MICROGRAMS_PER_CUBIC_METER,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key=SENSOR_KIND_TEMPERATURE,
+        name="Temperature",
+        device_class=SensorDeviceClass.TEMPERATURE,
+        native_unit_of_measurement=TEMP_CELSIUS,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key=SENSOR_KIND_VOC,
+        name="VOC",
+        device_class=SensorDeviceClass.VOLATILE_ORGANIC_COMPOUNDS,
+        native_unit_of_measurement=CONCENTRATION_PARTS_PER_MILLION,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up AirVisual sensors based on a config entry."""
+    coordinator: DataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+
+    async_add_entities(
+        AirVisualProSensor(coordinator, description)
+        for description in SENSOR_DESCRIPTIONS
+    )
+
+
+class AirVisualProSensor(AirVisualProEntity, SensorEntity):
+    """Define an AirVisual Pro sensor."""
+
+    _attr_has_entity_name = True
+
+    @property
+    def measurements(self) -> dict[str, Any]:
+        """Define collected measurements data."""
+        return self.coordinator.data["measurements"]
+
+    @callback
+    def _async_update_from_latest_data(self) -> None:
+        """Update the entity from the latest data."""
+        if self.entity_description.key == SENSOR_KIND_AQI:
+            if self.coordinator.data["settings"]["is_aqi_usa"]:
+                self._attr_native_value = self.measurements["aqi_us"]
+            else:
+                self._attr_native_value = self.measurements["aqi_cn"]
+        elif self.entity_description.key == SENSOR_KIND_BATTERY_LEVEL:
+            self._attr_native_value = self.coordinator.data["status"]["battery"]
+        elif self.entity_description.key == SENSOR_KIND_CO2:
+            self._attr_native_value = self.measurements.get("co2")
+        elif self.entity_description.key == SENSOR_KIND_HUMIDITY:
+            self._attr_native_value = self.measurements.get("humidity")
+        elif self.entity_description.key == SENSOR_KIND_PM_0_1:
+            self._attr_native_value = self.measurements.get("pm0_1")
+        elif self.entity_description.key == SENSOR_KIND_PM_1_0:
+            self._attr_native_value = self.measurements.get("pm1_0")
+        elif self.entity_description.key == SENSOR_KIND_PM_2_5:
+            self._attr_native_value = self.measurements.get("pm2_5")
+        elif self.entity_description.key == SENSOR_KIND_TEMPERATURE:
+            self._attr_native_value = self.measurements.get("temperature_C")
+        elif self.entity_description.key == SENSOR_KIND_VOC:
+            self._attr_native_value = self.measurements.get("voc")
```

Comment:
I think you could replace this whole section with a dict lookup of `KEY_TO_MEASUREMENT` to avoid the linear search